lib/grant-types/implicit-grant-type.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+var AbstractGrantType = require('./abstract-grant-type');
+var InvalidArgumentError = require('../errors/invalid-argument-error');
+var Promise = require('bluebird');
+var util = require('util');
+
+/**
+ * Constructor.
+ */
+
+function ImplicitGrantType(options) {
+  options = options || {};
+
+  if (!options.model) {
+    throw new InvalidArgumentError('Missing parameter: `model`');
+  }
+
+  if (!options.model.saveToken) {
+    throw new InvalidArgumentError('Invalid argument: model does not implement `saveToken()`');
+  }
+
+  if (!options.user) {
+    throw new InvalidArgumentError('Missing parameter: `user`');
+  }
+
+  this.scope = options.scope;
+  this.user = options.user;
+
+  AbstractGrantType.call(this, options);
+}
+
+/**
+ * Inherit prototype.
+ */
+
+util.inherits(ImplicitGrantType, AbstractGrantType);
+
+/**
+ * Handle implicit token grant.
+ */
+
+ImplicitGrantType.prototype.handle = function(request, client) {
+  if (!request) {
+    throw new InvalidArgumentError('Missing parameter: `request`');
+  }
+
+  if (!client) {
+    throw new InvalidArgumentError('Missing parameter: `client`');
+  }
+
+  return this.saveToken(this.user, client, this.scope);
+};
+
+/**
+ * Save token.
+ */
+
+ImplicitGrantType.prototype.saveToken = function(user, client, scope) {
+  var fns = [
+    this.validateScope(user, client, scope),
+    this.generateAccessToken(),
```

Comment:
This should be invoked with `client, user, scope` to match the signature of `AbstractGrantType.prototype.generateAccessToken`

Reply:
This would pass any tests that are implemented where generateAccessToken isn't implemented on the model, or where client, user, or scope is not used within that function

Reply:
Nice catch. Thanks, @fabiancook!

Reply:
No issues! Happened to be implementing a server needing the implicit grant, tested this branch against that server, works perfect with this change :)